myning/utils/species_rarity.py

```diff
@@ -39,6 +39,12 @@
 
 SPECIES_WEIGHTS = [150, 75, 40, 20, 10, 7, 4]
 
+def get_current_tier() -> None:
+    player = Player()
+    highest_tier= max(map(lambda x: x.rarity_tier , player.discovered_species))
```

Comment:
This could result in getting a species you haven't discovered yet. For example, if you've unlocked Dragonborn but not Tiefling, that would put your roll into a high enough tier to get a Tiefling even if you haven't discovered that race. I think we just need a filter on the method in player to only have discovered species in the list of possibilities.

Reply:
Great catch! I'll add add a filter to the `roll_for_species` or loop through til you find one that's in your discovered list. What do you think about removing the chance of being a Human? Through my testing that happened vast majority of the time until I had a massive boost to my research.